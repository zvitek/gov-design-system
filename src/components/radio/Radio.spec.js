import { shallowMount } from '@vue/test-utils'
import GovRadio from '@components/radio/Radio'

let wrapper

describe('GovRadio', () => {
    beforeEach(() => {
        wrapper = shallowMount(GovRadio)
    })

    it('is called', () => {
        expect(wrapper.name()).toBe('GovRadio')
        expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('render correctly', () => {
        const wrapper = shallowMount(GovRadio, {
            propsData: {
                customUniqueId: 'radio_abc'
            }
        })
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('has an input radio', () => {
        expect(wrapper.contains('input[type=radio]')).toBeTruthy()
    })

    it('emit input event with value when value change', () => {
        wrapper.setProps({ value: true })
        expect(wrapper.vm.computedValue).toBeTruthy()
        wrapper.vm.computedValue = false
        const valueEmitted = wrapper.emitted()['input'][0]
        expect(valueEmitted).toContainEqual(false)
    })

    it('method focus() gives focus to the input element', (done) => {
        wrapper.vm.$refs.input.focus = jest.fn()
        wrapper.vm.focus()
        wrapper.vm.$nextTick(() => {
            expect(wrapper.vm.$refs.input.focus).toHaveBeenCalled()
            done()
        })
    })
})
