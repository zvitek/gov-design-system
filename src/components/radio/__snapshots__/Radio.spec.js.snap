// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GovRadio render correctly 1`] = `<li class="gov-form-group__item"><input type="radio" class="gov-form-control__radio" value="false" id="radio_abc"> <label class="gov-form-control__label" for="radio_abc"></label> <span class="gov-form-control__indicator"></span></li>`;
